### ISSUE TITLE
the WF engine should resolves escaped variables

### ISSUE THREAD

**Issue by FabienArcellier** [writer/writer-framework#513]
The `WF` engine no longer resolves escaped variables `@{a\.b}` to point to a state encoded with a `.` as a separator:

```python
state = wf.init_state({
    "a.b": 2,
})
```
![image](https://github.com/user-attachments/assets/dec5113a-87e7-4882-a0fa-3ff4e0e3ba24)

---

| feature | status |
|---------- | -------- |
| the engine resolve the escape variable in text field | KO |
| the engine resolve the escape binding | KO |
| the autocompletion handle escape on "a.b" | KO | 
| the state explorer show the variable | OK |
| `WriterState` should support field mapping to handle dot notation | KO | 

#### WriterState should support field mapping to handle dot notation

```python
class MyState(wf.WriterState):
    counterA: int = wf.field(key='counter.a', default=0)
    counterB: int = wf.field(key='counter.b', default=0)

    @wf.property(['counter\.a', 'counter\.b'])
    # alternative : @wf.property(['counterA', 'counterB'])
    def counter_sum(self) -> int:
        return self.counterA + self.counterB # equivalent of self['counter.a'] + self['counter.b']
```

**Comment by FabienArcellier** [comment]
implement in https://github.com/writer/writer-framework/pull/568

**Comment by FabienArcellier** [comment]
it's available in [release 0.8.1](https://pypi.org/project/writer/)
